### PR TITLE
Reorganize query tests and integration assets

### PR DESCRIPTION
**Pull request by birkskyum**
The integration test folder was quite bloated by a mix of assets and test suites. I moved all the assets to an asset folder, that now contains:

integration / assets / 
- data
- geojson
- glyphs
- image
- sprites
- styles
- tiles
- tilesets
- video

Also, the files related to the test-query suite test are now in "integration/query".

### CODE REVIEW

File: build/generate-query-test-fixtures.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-const rootFixturePath = 'test/integration/';
-const suitePath = 'query-tests';
+const rootFixturePath = 'test/integration/query/';
+const suitePath = 'tests';
 
-import {generateFixtureJson} from '../test/integration/lib/generate-fixture-json.js';
+import {generateFixtureJson} from '../test/integration/query/generate-fixture-json.js';
```
Comment (HarelM) [inline build/generate-query-test-fixtures.ts:4]: Isn't this suppressed to be .ts?
Also, why do we need this file? Is the other file referenced somewhere m

File: src/data/bucket/symbol_bucket.test.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {RGBAImage} from '../../util/image';
 import {ImagePosition} from '../../render/image_atlas';
 import {IndexedFeature, PopulateParameters} from '../bucket';
 import {StyleImage} from '../../style/style_image';
-import glyphs from '../../../test/unit/assets/fontstack-glyphs.json';
+import glyphs from '../../../test/integration/assets/glyphs/fontstack-glyphs.json';
```
Comment (HarelM) [inline src/data/bucket/symbol_bucket.test.ts:19]: The reference to the integration folder is not great, can we use hard coded json content in this test, hopefully a lot simpler than the one in the file?

File: build/generate-query-test-fixtures.ts (hunk 1 above)
Comment (birkskyum) [reply build/generate-query-test-fixtures.ts]: Right, it's ts, not js, and the referenced file is only used here, so I have merged the two files now.

File: src/data/bucket/symbol_bucket.test.ts (hunk 2 above)
Comment (birkskyum) [reply src/data/bucket/symbol_bucket.test.ts]: I copied the file, so now the integration and unit tests don't share assets which is more clean. I hoped it was possible to reduce the filesize by removing elements of the glyph array, but then the unit tests break.

Comment (HarelM) [reply src/data/bucket/symbol_bucket.test.ts]: Ok. Thanks for trying!
That'll do I guess.

**Review by HarelM — APPROVED** [review]
Please check the CI logs to see that all the tests are running (I.e that the same number of tests are executed)

**Comment by birkskyum** [comment]
> Please check the CI logs to see that all the tests are running (I.e that the same number of tests are executed)

Yep, we still got this:
**Render**
1209 passed (99.1%)
2 passed but were ignored (0.2%)
9 ignored (0.7%)

**Expression**
358 passed (100.0%)

**Query**
Tests:       127 passed, 127 total

**Comment by wipfli** [comment]
#955 removes the jsdoc warnings that are shown in the "files" section of this pull request.

**Pull request merged**